archiver/archive.go

```diff
@@ -0,0 +1,143 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package archiver
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/joeshaw/multierror"
+
+	"github.com/pkg/errors"
+)
+
+// PackageProperties defines properties describing the package. The structure is used for archiving.
+type PackageProperties struct {
+	Name    string
+	Version string
+	Path    string
+}
+
+// ArchivePackage method builds and streams an archive with package content.
+func ArchivePackage(w io.Writer, properties PackageProperties) (err error) {
+	gzipWriter := gzip.NewWriter(w)
+	tarWriter := tar.NewWriter(gzipWriter)
+	defer func() {
+		var me multierror.Errors
+
+		if err != nil {
+			me = append(me, err)
```

Comment:
Nit: I stumbled over `me` on what it is. Perhaps use `multiErr` instead?

Reply:
Fixed.